controllers/dockermachine_controller.go

```diff
@@ -115,6 +115,12 @@ func (r *DockerMachineReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 		return ctrl.Result{}, err
 	}
 
+	//delete dockerMachine
+	if !machine.ObjectMeta.DeletionTimestamp.IsZero() {
+		log.Info("Start delete dockerMachine")
```

Comment:
There should be a check to ensure the finalizer is present before calling `r.delete()`, otherwise return early.